ros/service_client.go

```diff
@@ -26,9 +27,14 @@ func newDefaultServiceClient(logger Logger, nodeId string, masterUri string, ser
 	client.srvType = srvType
 	client.masterUri = masterUri
 	client.nodeId = nodeId
+	client.SetDeadline(10)
 	return client
 }
 
+func (c *defaultServiceClient) SetDeadline(ms uint32) {
```

Comment:
If you call it "deadline" it must be an absolute value (time.Time). That way it will be consistent with "net" and its notion of deadlines.

What you have here is a _timeout_. I think, it is better expressed directly as time.Duration